openmessaging-transport-binding.md

```diff
@@ -0,0 +1,289 @@
+
+This mapping is intentionally robust against changes, including the addition 
+and removal of event attributes, and also accommodates vendor extensions to the 
+event metadata. Any mention of event attributes other than contentType and data 
+is exemplary.
+
+
+### 2.1. contentType Attribute
+
+The `contentType` attribute is assumed to contain a media-type expression
+compliant with [RFC2046][RFC2046].
+
+### 2.2. data Attribute
+
+The `data` attribute is assumed to contain opaque application data that is
+encoded as declared by the `contentType` attribute.
+
+An application is free to hold the information in any in-memory representation
+of its choosing, but as the value is transposed into OpenMessaging as defined 
+in this specification, the assumption is that the `data` attribute value is made
+available as a sequence of bytes.
+
+For instance, if the declared `contentType` is
+`application/json;charset=utf-8`, the expectation is that the `data` attribute
+value is made available as [UTF-8][RFC3629] encoded JSON text for use in
+OpenMessaging.
+
+
+## 3. OpenMessaging Message Mapping
+
+With OpenMessaging, the content mode is chosen by the sender of the event. 
+Protocol usage patterns that might allow solicitation of events using a 
+particular content mode might be defined by an application, but are not 
+defined here.
+
+The receiver of the event can distinguish between the two content modes by
+inspecting the`contentType` property  in the *properties* of the OpenMessaging 
+message. If the value is prefixed with the CloudEvents media type 
+application/cloudevents, 
+indicating the use of a known event format, the receiver uses structured mode, 
+otherwise it defaults to binary mode.
+
+If a receiver detects the CloudEvents media type, but with an event format that 
+it cannot handle, for instance application/cloudevents+avro, it MAY still treat 
+the event as binary and forward it to another party as-is.
+
+
+### 3.1. Binary Content Mode
+
+The *binary* content mode accommodates any shape of event data, and allows for
+efficient transfer and without transcoding effort.
+
+#### 3.1.1. OpenMessaging contentType
+
+For the *binary* mode, the  `contentType`  property field value maps directly to
+the CloudEvents `contentType` attribute.
+
+#### 3.1.2. Event Data Encoding
+
+The [`data` attribute](#22-data-attribute) byte-sequence is used as the 
+[OpenMessaging data][OpenMessaging-Spec] section.
+
+#### 3.1.3. Metadata Headers
+
+All [CloudEvents][CE] attributes with exception of `contentType` and `data`
+MUST be individually mapped to and from the *properties* section.
+
+##### 3.1.3.1 OpenMessaging Properties Names
+
+
+Cloud Event attributes are prefixed with "cloudEvents:" for use in the
+[properties][OpenMessaging-Spec] section.
+
+Examples:
+
+* `eventTime` maps to `cloudEvents:eventTime`
+* `eventID` maps to `cloudEvents:eventID`
+* `cloudEventsVersion` maps to `cloudEvents:cloudEventsVersion`
```

Comment:
would these appear under "headers" or "properties" in the OM message?

Reply:
oh, never mind, in the example I see they're under "properties". This doc should mandate this with RFC2119 language.